doc/users-guide/base/configs.rst

```diff
@@ -0,0 +1,220 @@
+``config`` prefix causes the compiler to add a command-line flag to
+the generated executable that supports overriding that default value.
+For example, if we run the program as follows:
+
+.. literalinclude:: examples/users-guide/base/configs.execopts
+  :language: text
+  :lines: 1
+
+then *n* will be initialized to the value 1000 for that execution of
+the program.  The arguments for these flags can also be specified
+using ``=`` (with no spaces) as the separator between the argument
+name and its value, as follows:
+
+.. literalinclude:: examples/users-guide/base/configs.execopts
+  :language: text
+  :lines: 2
+
+There is no difference between these two forms—it's simply a matter of
+taste.
+
+As with normal variables, a variable declared as a ``config var`` can
+be re-assigned throughout its lifetime.  In contrast, ``config const``
+declarations cannot be changed once they are initialized, whether that
+initial value comes from the source code expression or the
+command-line.  The following statement declares a pair of configurable
+constants:
+
+.. literalinclude:: examples/users-guide/base/configs.chpl
+  :language: chapel
+  :lines: 2-3
+
+As with the previous case, we can override their default values on the
+executable's command-line.  For instance the following switches to a
+smaller value of *epsilon* and flips *debug* to ``true``:
+
+.. literalinclude:: examples/users-guide/base/configs.execopts-fake
+  :language: text
+  :lines: 3
+
+Boolean ``config`` declarations, like *debug* above, also support a
+mode in which they can be set to ``true`` simply by using the flag
+without a supporting value.  Thus, the previous command-line could
+also have been written:
+
+.. literalinclude:: examples/users-guide/base/configs.execopts-fake
+  :language: text
+  :lines: 4
+
+
+``config param`` and ``config type``
+------------------------------------
+
+Chapel also supports configurable ``param`` and ``type`` declarations
+via command-line flags.  However, since ``param`` values and types
+must be known to the compiler, the default values are overridden using
+flags on the compiler rather than the generated executable.
+
+As an example, the following statements declare a series of
+configurable ``param`` and ``type`` symbols:
+
+.. literalinclude:: examples/users-guide/base/configs2.chpl
+  :language: chapel
+  :lines: 3-7
+
+When overriding ``config`` declarations in the compiler the ``-s``
+flag is used to set the new value.  As an example, to set **verbose**
+to ``true``, we could compile as follows:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
+  :language: text
+  :lines: 1
+
+As with the generated executable, boolean values can also be set on
+the compiler's command line simply using the name of the flag.  Thus,
+the following would be equivalent to the previous line:
+
+.. literalinclude:: examples/users-guide/base/configs2.compopts-fake
+  :language: text
+  :lines: 2
+
```

Comment:
Would be nice to show also that a space after "-s" is allowed, e.g. "chpl -s size=8" .